php/media/class-gallery.php

```diff
@@ -398,6 +391,39 @@ protected function register_settings() {
 		$media_settings->add_setting( $this->settings );
 	}
 
+	/**
+	 * Inits the cloudinary gallery using block attributes.
+	 *
+	 * @param string $content The post content.
+	 * @param array  $block   Block data.
+	 *
+	 * @return string
+	 */
+	public function prepare_block_render( $content, $block ) {
+		if ( 'cloudinary/gallery' !== $block['blockName'] ) {
+			return $content;
+		}
+
+		$attributes                = Utils::expand_dot_notation( $block['attrs'], '_' );
+		$attributes                = array_merge( self::$default_config, $attributes );
+		$attributes['mediaAssets'] = $attributes['selectedImages'];
+		$attributes['cloudName']   = $this->media->plugin->components['connect']->get_cloud_name();
+		unset( $attributes['selectedImages'], $attributes['customSettings'] );
+
+		ob_start();
+		?>
+		<script>
+			window.addEventListener( 'load', function () {
+				var attributes = <?php echo wp_json_encode( $attributes ); ?>;
+				attributes.container = '.' + attributes.container;
+				cloudinary.galleryWidget( attributes ).render();
```

Comment:
Could we add a check here that ensures that `cloudinary.galleryWidget` is defined and available just to prevent any errors in the browser console if the library isn't loaded for some reason?